### PR TITLE
Maintenance: Refactor per code standards

### PR DESCRIPTION
**Pull request by aidvu**
Refactoring gapps to make `lint` happy. Just maintenance pickup from https://github.com/Automattic/wp-calypso/pull/14478

Test:
- Go to Domain Management
- Click Email tab on top
- Try adding a Google Apps subscription
- Try adding multiple subscriptions for different domains
- Make sure all tracking events are fired
- Make sure checkout finished successfully

### CODE REVIEW

File: client/components/upgrades/google-apps/google-apps-dialog/index.jsx [hunk 1]
```diff
@@ -11,41 +13,43 @@ import { cartItems } from 'lib/cart-values';
...
 		productsList: React.PropTypes.object.isRequired,
 		onAddGoogleApps: React.PropTypes.func.isRequired,
 		onClickSkip: React.PropTypes.func.isRequired,
 		onGoBack: React.PropTypes.func,
 		analyticsSection: React.PropTypes.string,
 		initialGoogleAppsCartItem: React.PropTypes.object
-	},
+	};
 
 	getInitialState() {
```
Comment (gwwar) [inline client/components/upgrades/google-apps/google-apps-dialog/index.jsx]: Like propTypes, we can use a class property here:

```jsx
static propTypes = {
    //...
};

state = {
    isAddingEmail: false,
    users: null,
   validationErrors: null,
};
```

File: client/components/upgrades/google-apps/google-apps-dialog/index.jsx [hunk 2]
```diff
@@ -86,24 +90,26 @@ const GoogleAppsDialog = React.createClass( {
 				onChange={ this.setUsers }
 				onBlur={ this.save } />
 		);
-	},
+	}
 
 	setUsers( users ) {
```
Comment (gwwar) [inline client/components/upgrades/google-apps/google-apps-dialog/index.jsx]: Handlers that we pass to other components/elements need be bound. React.createClass did this automatically, but here we need to do:
```jsx
setUsers = ( users ) => {
    //...
};
```

Comment (aidvu) [inline client/components/upgrades/google-apps/google-apps-dialog/index.jsx]: 🤦‍♂️  I thought about it when I was doing it, and then forgot to do it. GAH!

File: client/components/upgrades/google-apps/google-apps-dialog/index.jsx [hunk 3]
```diff
@@ -86,24 +90,26 @@ const GoogleAppsDialog = React.createClass( {
...
 		this.setState( { users: users } );
-	},
+	}
 
 	save() {
```
Comment (gwwar) [inline client/components/upgrades/google-apps/google-apps-dialog/index.jsx]: Similarly, this should be:
```jsx
save = () => {
   //...
};
```

Comment (aidvu) [inline client/components/upgrades/google-apps/google-apps-dialog/index.jsx]: Yeah, I'll update all that use `this.`.

File: client/components/upgrades/google-apps/google-apps-dialog/index.jsx [hunk 4]
```diff
@@ -172,51 +183,53 @@ const GoogleAppsDialog = React.createClass( {
...
 		event.preventDefault();
 
-		this.recordEvent( 'addEmailButtonClick', this.props.analyticsSection );
+		this.props.recordAddEmailButtonClick( this.props.analyticsSection );
 
 		this.setState( { isAddingEmail: true } );
-	},
+	}
 
 	handleFormSubmit( event ) {
```
Comment (gwwar) [inline client/components/upgrades/google-apps/google-apps-dialog/index.jsx]: Also needs to be bound:
```jsx
handleFormSubmit = ( event ) => {
```

File: client/components/upgrades/google-apps/google-apps-dialog/index.jsx [hunk 5]
```diff
@@ -172,51 +183,53 @@ const GoogleAppsDialog = React.createClass( {
 				</button>
 			</footer>
 		);
-	},
+	}
 
 	handleAddEmail( event ) {
```
Comment (gwwar) [inline client/components/upgrades/google-apps/google-apps-dialog/index.jsx]:
```jsx
handleAddEmail = ( event ) => {
//...
}
```

File: client/components/upgrades/google-apps/google-apps-dialog/users.jsx [hunk 6]
```diff
@@ -31,39 +35,47 @@ const GoogleAppsUsers = React.createClass( {
 			lastName: { value: '', error: null },
 			domain: { value: this.props.domain, error: null }
 		};
-	},
+	}
 
 	render() {
 		const fields = this.props.fields || this.getInitialFields(),
-			allUserInputs = fields.map( this.inputsForUser );
+			allUserInputs = fields.map( this.inputsForUser ),
```
Comment (gwwar) [inline client/components/upgrades/google-apps/google-apps-dialog/users.jsx]: Optional, but I do prefer one `const` or `let` per declaration. eg
```
const fields = //...;
const allUserInputs = //...;
const { translate } = this.props;
```

File: client/components/upgrades/google-apps/google-apps-dialog/users.jsx [hunk 7]
```diff
@@ -31,39 +35,47 @@ const GoogleAppsUsers = React.createClass( {
...
 		);
-	},
+	}
 
 	fieldClasses( fieldName ) {
 		return `google-apps-dialog__user-field google-apps-dialog__user-${ fieldName }`;
-	},
+	}
 
 	inputsForUser( user, index ) {
```
Comment (klimeryk) [inline client/components/upgrades/google-apps/google-apps-dialog/users.jsx]: Gotta bind this one too - see line 42.

**Comment by aidvu** [comment]
Left bind for next iteration. Should also finish it today.

**Pull request merged**